### ISSUE TITLE
[5.2.0] Release Checklist

### ISSUE THREAD

**Issue by rahulporuri** [enthought/apptools#293]
We would like to make an apptools release to ship the `on_trait_change` to `observe` changes to end users.

Target date: Undecided

Release Candidate by Undecided

Release by Undecided

Note : I have removed the "Release Candidate" section from the release checklist for this release. It's a small release and it's not worth a release candidate.

Release blockers
----------------
- [ ] (add blocking issues/PRs here)

Pre-release
---
- [ ] Backport PRs that have been merged to master to the maintenance branch. Use the "need backport ..." tag if there is one (but don't rely 100% on it)
- [ ] Verify that no other open issue needs to be addressed before the release.
- [ ] Test against other ETS packages and other ETS-using projects
- [ ] Check MANIFEST, requirements, changelog are still up to date.
- [ ] Test building the documentation

Release
-------
- [ ] Create branch `release/<release version number>` from `maint/<release version number>` branch.
  - [ ] Set release to `<release version number>`, and set `IS_RELEASED` is true; commit
  - [ ] Install from source distribution and run tests again
  - [ ] Open a PR against `maint/<release version number>` with this being the last commit so that CI is built on the release commit
  - [ ] Once CI is done building merge PR
  - [ ] Bump the micro version number i.e. `<release version number + 0.0.1>` and set `IS_RELEASED` to false; commit.
  - [ ] Open and then merge a separate PR against `maint/<release version number>`
- [ ] From the commit at which `IS_RELEASED` is true and version is `<release version number>`, tag (annotated!) `git tag -a -m "Release <release version number>" <release version number> <commit-hash>`
- [ ] Push the tag `git push origin <release version number>`
- [ ] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting index.html looks good.
- [ ] Upload to PyPI
- [ ] Test the PyPI package

Post-release
-------------
- [ ] Package update for `enthought/free` repository (for EDM)
- [ ] Backport change log (and any deletion of news fragments) to master
- [ ] Update GitHub Release pages https://github.com/enthought/apptools/releases
- [ ] Announcement (e.g. Google Group)
